File: abc/abc157/abc157_b.cs
// https://atcoder.jp/contests/abc157/submissions/10470543

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var A = new int[3][];
            for (int i = 0; i < 3; i++)
            {
                A[i] = Console.ReadLine().Split().Select(int.Parse).ToArray();
            }

            var N = int.Parse(Console.ReadLine());

            var b = new int[N];
            for (int i = 0; i < N; i++)
            {
                b[i] = int.Parse(Console.ReadLine());
            }

            var C = new bool[3][];
            for (int i = 0; i < 3; i++)
            {
                C[i] = new bool[3];
            }

            for (int i = 0; i < N; i++)
            {
                for (int j = 0; j < 3; j++)
                {
                    for (int k = 0; k < 3; k++)
                    {
                        if (A[j][k] == b[i])
                        {
                            C[j][k] = true;
                        }
                    }
                }
            }

            var result = false;
            for (int i = 0; i < 3; i++)
            {
                if ((C[i][0] && C[i][1] && C[i][2]) || (C[0][i] && C[1][i] && C[2][i]))
                {
                    result = true;
                }
            }
            if ((C[0][0] && C[1][1] && C[2][2]) || (C[2][0] && C[1][1] && C[0][2]))
            {
                result = true;
            }

            Console.WriteLine($"{(result ? "Yes" : "No")}");
        }
    }
}

File: abc/abc157/abc157_b.fs
// https://atcoder.jp/contests/abc157/submissions/11872711

open System.Collections.Generic

let A = new List<int []>()

for i in [ 1..3 ] do
    stdin.ReadLine().Split()
    |> Array.map int
    |> A.Add

let N = stdin.ReadLine() |> int

let b = new List<int>()

for i in [ 1..N ] do
    stdin.ReadLine()
    |> int
    |> b.Add

let mutable C = new List<bool []>()

for i in [ 1..3 ] do
    [| false; false; false |] |> C.Add

for i in [ 0..N - 1 ] do
    for j in [ 0..2 ] do
        for k in [ 0..2 ] do
            if A.[j].[k] = b.[i] then C.[j].[k] <- true

let mutable result = false

for i in [ 0..2 ] do
    if C.[i].[0] && C.[i].[1] && C.[i].[2] then result <- true
    if C.[0].[i] && C.[1].[i] && C.[2].[i] then result <- true
if C.[0].[0] && C.[1].[1] && C.[2].[2] then result <- true
if C.[2].[0] && C.[1].[1] && C.[0].[2] then result <- true

if result then "Yes"
else "No"
|> stdout.WriteLine
